src/Uns.ts

```diff
@@ -259,16 +261,20 @@ export default class Uns extends NamingService {
   }
 
   async registryAddress(domain: string): Promise<string> {
-    if (!this.checkDomain(domain)) {
-      throw new ResolutionError(ResolutionErrorCode.UnsupportedDomain, {
-        domain,
-      });
-    }
+    let namehash: string;
+    if (domain.startsWith('0x')) {
+      namehash = this.namehash(domain);
+    } else {
+      if (!this.checkDomain(domain)) {
```

Comment:
Could be simplified to `else if {`

Reply:
By doing so the statement becomes as follow:
<img width="558" alt="Screen Shot 2021-06-30 at 5 45 55 PM" src="https://user-images.githubusercontent.com/16522024/124048435-3377a680-d9cb-11eb-8895-6d0e8cda37c3.png">

I really don't like the trailing else statement so I left it as is
<img width="543" alt="Screen Shot 2021-06-30 at 5 46 18 PM" src="https://user-images.githubusercontent.com/16522024/124048468-42f6ef80-d9cb-11eb-9c19-8873a5f23ff6.png">